docs/page-objects/rules.rst

```diff
@@ -1,782 +1,166 @@
-.. _rules-intro:
+.. _rules:
 
-Apply rules
-===========
+=====
+Rules
+=====
 
-Basic Usage
------------
+Rules are :class:`~.ApplyRule` objects that tell web-poet which :ref:`page
+object class <page-object-classes>` to use for a given combination of input URL
+and :ref:`item type <item-types>`.
 
-@handle_urls
-~~~~~~~~~~~~
+Defining rules
+==============
 
-web-poet provides a :func:`~.handle_urls` decorator, which allows to
-declare how a page object can be used (applied):
+The :func:`~.handle_urls` decorator is the simplest way to define a rule for
+a page object. For example:
 
-* for which websites / URL patterns it works,
-* which data type (item classes) it can return,
-* which page objects it can replace (override; more on this later).
+.. _handle_url_example:
 
 .. code-block:: python
 
     from web_poet import ItemPage, handle_urls
+
     from my_items import MyItem
 
     @handle_urls("example.com")
     class MyPage(ItemPage[MyItem]):
-        # ...
-
-
-``handle_urls("example.com")`` can serve as documentation, but it also enables
-getting the information about page objects programmatically.
-The information about all page objects decorated with
-:func:`~.handle_urls` is stored in ``web_poet.default_registry``, which is
-an instance of :class:`~.RulesRegistry`. In the example above, the
-following :class:`~.ApplyRule` is added to the registry:
-
-.. code-block::
-
-    ApplyRule(
-        for_patterns=Patterns(include=('example.com',), exclude=(), priority=500),
-        use=<class 'MyPage'>,
-        instead_of=None,
-        to_return=<class 'my_items.MyItem'>,
-        meta={}
-    )
-
-Note how ``rule.to_return`` is set to ``MyItem`` automatically.
-Such rules can be used by libraries like `scrapy-poet`_. For example,
-if a spider needs to extract ``MyItem`` from some page on the ``example.com``
-website, `scrapy-poet`_ now knows that ``MyPage`` page object can be used.
-
-.. _scrapy-poet: https://scrapy-poet.readthedocs.io
-
-Specifying the URL patterns
-~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-:func:`~handle_urls` decorator uses url-matcher_ library to define the
-URL rules. Some examples:
-
-.. code-block:: python
-
-    # page object can be applied on any URL from the example.com domain,
-    # or from any of its subdomains
-    @handle_urls("example.com")
-
-    # page object can be applied on example.com pages under /products/ path
-    @handle_urls("example.com/products/")
-
-    # page object can be applied on any URL from example.com, but only if
-    # it contains "productId=..." in the query string
-    @handle_urls("example.com?productId=*")
```

Comment:
I may be wrong, but it seems that having these basic examples cover almost all use cases, without asking users to check the external docs (which e.g. talk about some proxies in the examples, not page objects).